src/synapso_core/summarizer/instruct_summarizer.py

```
@@ -58,6 +69,7 @@ def _prepare_prompt(self, question: str, results: List[Tuple[str, float]]) -> st
 
         # 5. Use apply_chat_template, which will correctly add the assistant prompt turn.
         # This is the most reliable way to format prompts.
+        assert self.tokenizer is not None
```

Ruff (0.12.2)

Use of `assert` detected

(S101)